File: emergencyBedApi/src/main/java/com/DouglasLessa/emergencyBedApi/model/domain/Hospital.java
package com.DouglasLessa.emergencyBedApi.model.domain;

import java.util.List;
import java.util.UUID;

import com.DouglasLessa.emergencyBedApi.model.security.User;

import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "hospitals")
public class Hospital {

    @Id
    private UUID id = UUID.randomUUID();

    private String name;
    private String phone;
    private String escalation;

    @Embedded
    @JoinColumn(name = "address_hospital_id")
    private Address address;

    private Integer totalCapacity;
    private Integer activeCapacity;
    private Integer numberOfBeds; 
    private Integer numberOfRooms;
    private Integer numberOfBuildings;
    private Integer numberOfFloors;

    @OneToMany(mappedBy = "hospital")
    private List<Bedroom> rooms;

    @OneToMany(mappedBy = "hospital")
    private List<Ambulance> ambulances;

    @OneToMany(mappedBy = "Users")
    private List<User> users;
}

File: go-api/internal/models/hospital.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// Hospital represents a hospital in the system
type Hospital struct {
	ID              uuid.UUID `json:"id" gorm:"type:uuid;primary_key;default:uuid_generate_v4()"`
	Name            string    `json:"name" gorm:"not null;unique"`
	Phone           string    `json:"phone" gorm:"not null"`
	Escalation      string    `json:"escalation" gorm:"not null"`
	Address         Address   `json:"address" gorm:"embedded"`
	TotalCapacity   int       `json:"totalCapacity" gorm:"not null"`
	ActiveCapacity  int       `json:"activeCapacity" gorm:"not null"`
	NumberOfBeds    int       `json:"numberOfBeds" gorm:"not null"`
	NumberOfRooms   int       `json:"numberOfRooms" gorm:"not null"`
	NumberOfBuildings int     `json:"numberOfBuildings" gorm:"not null"`
	NumberOfFloors  int       `json:"numberOfFloors" gorm:"not null"`
	CreatedAt       time.Time `json:"createdAt" gorm:"autoCreateTime"`
	UpdatedAt       time.Time `json:"updatedAt" gorm:"autoUpdateTime"`
} 